astropy/cosmology/core.py

```diff
@@ -149,44 +175,127 @@ def format_spec(self):
         """String format specification."""
         return self._fmt
 
+    @property
+    def fixed(self):
+        """Whether the Parameter is fixed; true parameters are not."""
+        return self._fixed
+
+    # -------------------------------------------
+    # 'property' descriptor overrides
+
+    @property
```

Comment:
Quite possibly. It's unlikely someone would mess with the attribute even if not protected, but modifying ``._fget``  is quite easy if one really needs to change ``fget``. But Parameter, like Cosmology, should be 'immutable'.

Reply:
I think a `property.fget` really should be immutable too - it is only accessible via the class and it seems obvious one should not change it. So it would seem to add complexity and lose speed for very little benefit.

Reply:
Perhaps ``Parameter`` should just inherit from `property`. Then things like ``fget`` would be automatically immutable.
``setter`` and ``deleter`` can just raise errors. 

```
class A:
    @property
    def attr(self):
        print(2)
a = A()
A.attr.fget = lambda self: print(3)  # fails
```

Reply:
Indeed, that means circling back a bit, but it does make sense...